GVFS/GVFS.Common/Tracing/JsonTracer.cs

```diff
@@ -83,13 +90,22 @@ public bool HasLogFileEventListener
 
         public void AddEventListener(EventListener listener)
         {
+            if (Volatile.Read(ref this.isDisposed) == DISPOSED_TRUE)

```

Comment:
This is the only place we have a `Volatile.XXX` operation - do other operations on this variable need to consider threading concerns (i.e. should all reads / writes to this variable be volatile)?

Reply:
All `isDisposed` operations are using `Volatile` or `Interlocked` in `JsonTracer`.

Reply:
Could we mark `this.isDisposed` as `volatile` instead of using `Volatile.Read`?  (That's what we've done elsewhere in the code where we need volatile reads, and I wasn't sure how/if `Volatile.Read` was different)

Reply:
You are right - I got mixed up with the dispose variables in different files. Thanks!

Reply:
> You are right - I got mixed up with the dispose variables in different files. Thanks!

I'm going to unresolve this comment as I'm still wondering about `volatile` vs `Volatile.Read`